iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/voter/FittingAffWordsMatchVoter.java

```diff
@@ -0,0 +1,157 @@
+        Preconditions.checkArgument(minFittingOrgWordsRatio > 0 && minFittingOrgWordsRatio <= 1);
+        Preconditions.checkArgument(minFittingWordSimilarity > 0 && minFittingWordSimilarity <= 1);
+        
+        
+        this.charsToFilter = charsToFilter;
+        this.minWordLength = minWordLength;
+        this.minFittingOrgWordsRatio = minFittingOrgWordsRatio;
+        this.minFittingWordSimilarity = minFittingWordSimilarity;
+    }
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Returns true if minFittingOrgWordsPercentage of {@link AffMatchOrganization#getName()} words
+     * are found in any section of {@link AffMatchAffiliation#getOrganizationName()}.
+     * 
+     * @see #FittingAffWordsMatchVoter(List, int, double, double)
+     */
+    @Override
+    public boolean voteMatch(AffMatchAffiliation affiliation, AffMatchOrganization organization) {
+        
+        List<String> affSections = organizationSectionsSplitter.splitToSections(affiliation.getOrganizationName());
+        
+        
+        String filteredOrgName = filterName(organization.getName());
+        
+        if (StringUtils.isEmpty(filteredOrgName)) {
+            return false;
+        }
+        
+        Set<String> orgWords = Sets.newHashSet(StringUtils.split(filteredOrgName));
+        
+        
+        for (String affSection : affSections) {
+            
+            String filteredAffSectionName = filterName(affSection);
+            
+            if (StringUtils.isEmpty(filteredAffSectionName)) {
+                continue;
+            }
+            
+            Set<String> affWords = Sets.newHashSet(StringUtils.split(filteredAffSectionName));
+            
+            if (voteSectionMatch(affWords, orgWords)) {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private boolean voteSectionMatch(Set<String> affSectionWords, Set<String> orgWords) {
+        int affWordsCount = affSectionWords.size();
+        int fittingWordsCount = 0;
+        
+        for (String affSectionWord : affSectionWords) {
+            if (wordsContainsSimilarWord(orgWords, affSectionWord)) {
+                ++fittingWordsCount;
+            }
+        }
+        
+        double fittingWordsRatio = (double)fittingWordsCount/affWordsCount;
+        
+        return fittingWordsRatio >= minFittingOrgWordsRatio;
+    }
+
+    private boolean wordsContainsSimilarWord(Set<String> affWords, String orgWord) {
+        
+        for (String affWord : affWords) {
+            double similarity = StringUtils.getJaroWinklerDistance(affWord, orgWord);
+            
+            if (similarity >= minFittingWordSimilarity) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private String filterName(String name) {
```

Comment:
I have seen this code somewhere else ;-) Isn't it duplicated? The same about wordsContainsSimilarWord.
I would move it to a separate simple services and test it separately.